test/distributed/test_c10d.py

```diff
@@ -2223,6 +2223,17 @@ def __init__(self):
     def forward(self, x):
         return F.softmax(self.embedding(x), dim=1)
 
+class MixedDenseSparseGradientModule(nn.Module):
+    def __init__(self):
+        super(MixedDenseSparseGradientModule, self).__init__()
+        self.fc = nn.Linear(10, 10)
+        self.embedding = nn.EmbeddingBag(10, 10, sparse=True)
+        self.relu = nn.ReLU()
+
+    def forward(self, x):
+        x = self.relu(self.fc(x.to(torch.float)))
+        return F.softmax(self.embedding(x.to(torch.long)), dim=1)
```

Comment:
Can you try putting the linear layer after embedding table. It will also make more sense logically, i.e., first look up embeddings from the table and then use those embeddings to run fwd and bwd. The following code works for me locally:


```python
class SparseGradientModule(nn.Module):
    def __init__(self):
        super(SparseGradientModule, self).__init__()
        self.embedding = nn.EmbeddingBag(10, 10, sparse=True)
        self.linear = nn.Linear(10, 10)

    def forward(self, x):
        return F.softmax(self.linear(self.embedding(x)), dim=1)
```

Reply:
updated the MixedDenseSparseModule